File: src/main/java/tlint/cli/TLint.java
package tlint.cli;

import com.google.gson.GsonBuilder;
import com.google.gson.JsonSyntaxException;
import com.google.gson.Gson;
import java.util.ArrayList;
import java.util.List;

public class TLint {
    public File file = new File();

    static TLint read(String json) {
        TLint lint = new TLint();

        try {
            GsonBuilder builder = new GsonBuilder();
            Gson gson = builder.create();
            lint.file = gson.fromJson(json, File.class);
        } catch (JsonSyntaxException e) {
            //
        }

        return lint;
    }

    public static class File {
        public List<Issue> errors = new ArrayList<>();
    }

    public static class Issue {
        public String source;
        public int line;
        public String message;
    }
}

File: src/main/kotlin/tlint/cli/TLint.kt
package tlint.cli

import com.google.gson.GsonBuilder
import com.google.gson.JsonSyntaxException
import java.util.ArrayList

class TLint {
    var file = File()

    class File {
        var errors: List<Issue> = ArrayList()
    }

    class Issue {
        var source: String? = null
        var line: Int = 0
        var message: String? = null
    }

    companion object {

        internal fun read(json: String): TLint {
            val lint = TLint()

            try {
                val builder = GsonBuilder()
                val gson = builder.create()
                lint.file = gson.fromJson(json, File::class.java)
            } catch (e: JsonSyntaxException) {
                //
            } catch (e: java.lang.IllegalStateException) {
                //
            }

            return lint
        }
    }
}